foundry.md

```diff
@@ -561,7 +641,62 @@ Utils
          </account>
 ```
 
- - selectors for cheat code functions.
+- `#setPrank NEWCALLER` will set the `<prank/>` subconfiguration for the given account.
+
+```k
+    syntax KItem ::= "#setPrank" Int [klabel(foundry_setPrank)]
+ // -----------------------------------------------------------
+    rule <k> #setPrank NEWCALLER => . ... </k>
+         <callDepth> CD </callDepth>
+         <caller> CL </caller>
+         <origin> OG </origin>
+         <prank>
+           <prevCaller> .Account => CL </prevCaller>
+           <prevOrigin> .Account => OG </prevOrigin>
+           <newCaller> _ => NEWCALLER </newCaller>
+           <active> false => true </active>
+           <depth> _ => CD </depth>
+           <singleCall> _ => false </singleCall>
+           ...
+         </prank>
+```
+
+- `#setPrankWithOrigin NEWCALLER NEWORIGIN` will set the `<prank/>` subconfiguration for the given account.
+
+```k
+    syntax KItem ::= "#setPrankWithOrigin" Int Int [klabel(foundry_setPrankWithOrigin)]
+ // -----------------------------------------------------------------------------------
+    rule <k> #setPrankWithOrigin NEWCALLER NEWORIGIN => . ... </k>
+         <callDepth> CD </callDepth>
+         <caller> CL </caller>
+         <origin> OG </origin>
```

Comment:
What happens if you start a prank with one already active? Maybe we can have `#setPrank ACCT => #setPrankWithOrigin ACCT .Account`, so that we only have one rule updating the prank sub-configuration.

Then we can also have a rule for `#setPrankWithOrigin ...` which handles the case where `<active> true </active>` (by going to a `#foundry_error "Prank already active"` state somehow?

Reply:
Similarly, if you end a prank with none active, we should go to an error state. Please also add tests of these cases, and add those tests to the `tests/foundry/exclude` list for now.

Reply:
> What happens if you start a prank with one already active? Maybe we can have `#setPrank ACCT => #setPrankWithOrigin ACCT .Account`, so that we only have one rule updating the prank sub-configuration.
> 
> Then we can also have a rule for `#setPrankWithOrigin ...` which handles the case where `<active> true </active>` (by going to a `#foundry_error "Prank already active"` state somehow?

Starting a prank while another one is active will return a Fail error `[FAIL. Reason: You have an active prank already.]`. Marking the test with `testFail` will make the test pass.